### PR TITLE
CFD-320 : Homepage : Activity stream should contain only trimmed text

### PR DESCRIPTION
**Pull request by nader77**
#526

### CODE REVIEW

**Comment by nader77** [comment]
Trimmed activity stream on the homepage:
selection_073

**Comment by nader77** [comment]
@zero2one PR is ready.

**Comment by nader77** [comment]
Update: This PR does the following
* Trim body content to 150 chars, strip tags and add '...' at the end in the activity stream of the homepage.
* Fix the parameters that determine  the content of the activity stream in each group.
* Hide "Show more" button fix.

File: capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 1]
```diff
@@ -187,7 +187,7 @@ function c4m_messages_group_info(Message $message) {
   // Add group info for activity stream on the homepage,
   // When the request is from RESTful, "drupal_is_front_page" won't work.
   $handler = og_purl_og_context_handler();
-  if ((drupal_is_front_page() && empty($handler) || (isset($_REQUEST['group']) && is_array($_REQUEST['group'])))) {
+  if (((drupal_is_front_page() || !empty($_REQUEST['homepage'])) && empty($handler) || (isset($_REQUEST['group']) && is_array($_REQUEST['group'])))) {
```
Comment (amitaibu) [inline capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: ``$_REQUEST`` should be ``$_GET``, no?

**Comment by nader77** [comment]
@amitaibu Comments fixed.

**Comment by zero2one** [comment]
@amitaibu @nader77 

I'm sorry but I'm not gonna approve this solution.
We need to avoid hard-coded display of the entities in the activity stream.

* Drupal has display modes for entities, we should use them. 
* The settings can be featured in the content type features.
* We use Display suite to define the display of the display modes.
* We should use the same approach everywhere. This makes it easier to maintain.

### Message entity display modes
2 different display modes:
* global activity
* group activity

Where 
* "global activity" uses the "global activity" display mode of the node entity to render the node teaser. 
* "group activity" uses the "group activity" display mode of the node entity to render the node teaser.

### Node entity display modes
2 different display modes:
* global activity : used in the activity streams outside groups (homepage).
* group activity : used in the activity streams inside groups (group dashboard).

**Comment by zero2one** [comment]
@amitaibu @nader77 

Parked for next sprint.

**Comment by zero2one** [comment]
See CFD-352

**Comment by nader77** [comment]
Update:
* Created 2 display modes ``activity_group`` && ``activity_global``
* Updated all the content features with the new display modes.
* Added the module [Smart trim](https://www.drupal.org/project/smart_trim) which gives us dynamic trimming, suffex for the trimming and a strip tags option:
selection_062

**Comment by nader77** [comment]
Homepage:
selection_064

Group dashboard: 
selection_065

**Comment by nader77** [comment]
@HelenaEksler PR is ready for code review.

**Comment by HelenaEksler** [comment]
@nader77 Looks good for me.
@zero2one PR is ready for your review.

**Comment by zero2one** [comment]
Can you do following changes:

## Change code
* Remove outputting the node title directly to the activity stream in code.
* Remove outputting the number of comments & views from the code.

## Add the node title activity display modes
* Add the node title to the activity display modes (Activity Group & Activity Global).
* Add the "Footer stats" (number of comments & views) to the activity display modes.
* Update css if necessary.

## How to add the title field & footer stats to the node display:

Open the display mode setting for the node you want to update.
Select a layout
c4m-display-01

Choose for "One column".
c4m-display-02

Save the settings
c4m-display-03

Once the settings are saved, a lot more fields will be available to add to the display.
c4m-display-04

Drag and drop the title field above the body.
Drag and drop the footer stats underneath the body.
c4m-display-05

Save the settings.

**Comment by nader77** [comment]
@zero2one Before i do the changes, Just wanna mention that the title is created in the message not in the code itself:
edit_c4m_insert_comment___capacity4more

Does this change anything or i should still delete it from here and create it in the display mode ?

**Comment by nader77** [comment]
Footer is created in the code.

**Comment by zero2one** [comment]
@nader77 

I know but we realy should use the rendered node.

It makes it clearer for everybody that the node is rendered with a certain display mode + the display of the node part is managed in the node iteself.

**Comment by nader77** [comment]
@zero2one No problem, Just making sure :)

**Comment by nader77** [comment]
Update:
Added title and footer stats to all content types, Removed from code:
article___capacity4more

In comments, I had to keep the title from the message because when i tried to create view modes and add the node's title, I found out that i have only the comment title, Couldn't add the node's title:
article___capacity4more

**Comment by nader77** [comment]
@zero2one @HelenaEksler 
Activity stream after all the changes, Please point out if there's some unwanted changes or errors, Thanks.

activity_stream

**Pull request merged**